sql/src/main/java/io/crate/memory/OffHeapMemoryManager.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.memory;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+
+import java.util.ArrayList;
+
+public final class OffHeapMemoryManager implements MemoryManager {
+
+    private final ArrayList<ByteBuf> allocatedBuffers = new ArrayList<>();
+
+    public OffHeapMemoryManager() {
+    }
+
+    @Override
+    public ByteBuf allocate(int capacity) {
+        // TODO: Should we have a limit that is lower than the maxDirectMemory, so that queries abort before hitting a "real" OOM?
```

Comment:
Sounds good, any issue/reason not to do it?

Reply:
Mainly the added complexity / unclear if it is required.

I tend towards not having it for now. Putting off-heap behind a experimental feature flag and then see how it behaves first before we iterate further on it.